app/services/concerns/xccdf/profile_rules.rb

```diff
@@ -24,20 +24,20 @@ def save_profile_rules
       def profile_rules
         @profile_rules ||= @op_profiles.flat_map do |op_profile|
           profile_id = profile_id_for(ref_id: op_profile.id)
-          rule_ids_for(ref_ids: op_profile.selected_rule_ids).map do |rule_id|
+          rule_ids_for(ref_ids: op_profile.selected_rule_ids).map do |entity_id|
             ::ProfileRule.find_or_initialize_by(
-              profile_id: profile_id, rule_id: rule_id
+              profile_id: profile_id, entity_id: entity_id
```

Comment:
In order to make this work while maintaining compatibility with the rest of the code, you'd need the polymorphic parent thingie I explained above.